### PR TITLE
FS-2193: Fix "Label in name" issue on form jsons

### PR DESCRIPTION
**Pull request by tferns**
### Label in name:

[FS-2193](https://digital.dclg.gov.uk/jira/browse/FS-2193): In order to meet the "Label in name" accessibility standard, changes have been made to the form runner which allows us to hide the title and provide html in the label.

- Change form runner jsons to match new functionality on our fork of digital-form-builder 
  - https://github.com/communitiesuk/digital-form-builder/pull/108
- Anytime a MultilineTextField has a hint & title, we now hide the title and add the title text as html to the start of the hint
- Screen readers will now read out the pseudo-title, hint, text area and word count if applicable after these changes

###  Other changes:

- Reverting https://github.com/communitiesuk/funding-service-design-frontend/commit/41765652629a34ee97122da8991c0feb88f80c32 in https://github.com/communitiesuk/funding-service-design-frontend/commit/fe81e8ba11a36fd99233eeb5359e2d65ae02f2ac
  - The new changes in this PR regress those changes, so they've been reverted as part of this PR
- [scripts/form_refactor_util.py](https://github.com/communitiesuk/funding-service-design-frontend/compare/FS-2193-label-in-name-form-jsons?expand=1#diff-24bd1e8fecd6c58f3e00fd7dfd237c8d37975b83da139986e343b93f503d684a)
  - Script used to migrate existing form jsons and apply these changes
  - Can be easily extended in the future if we have other mass changes to form jsons
- [tests/test_form_refactor_util.py](https://github.com/communitiesuk/funding-service-design-frontend/compare/FS-2193-label-in-name-form-jsons?expand=1#diff-7cb47f71e1d511519944af64b4d9f4d833fc25dc539d0de74899eeb5e7e18a7d)
  - Generic test for the script mentioned above 
  - Works based on `input/expected` json files in `tests/form_refactor_util` directory
    - See [tests/form_refactor_util/fs-2193-place_title_in_hint_as_html-input.json](https://github.com/communitiesuk/funding-service-design-frontend/compare/FS-2193-label-in-name-form-jsons?expand=1#diff-a9352ea167f2b70345342c6196a2bde166808492ca0daf87bce4083db8caf4a0)
    - & [tests/form_refactor_util/fs-2193-place_title_in_hint_as_html-expected.json](https://github.com/communitiesuk/funding-service-design-frontend/compare/FS-2193-label-in-name-form-jsons?expand=1#diff-c450a85d7691fa3b87b8444f242ef8f52e5ec0cb45a9a51782571a97dd10d552)

###  Examples:

Before video:
https://user-images.githubusercontent.com/117724519/214770041-0e3121b8-96e9-4cea-b73a-587a9b930546.mp4

After video:
https://user-images.githubusercontent.com/117724519/214770064-a8957c9e-8282-44cb-8566-6afdcb7d5574.mp4

HTML: https://github.com/communitiesuk/funding-service-design-frontend/pull/249#issuecomment-1406343324
 
 
### Notes:

The [JIRA](https://digital.dclg.gov.uk/jira/browse/FS-2193) & [accessibility report document](https://digital.dclg.gov.uk/jira/secure/attachment/118150/DAC%20Accessibility%20Report%20WCAG%202.1%20for%20DLUHC%20Community%20Ownership%20Access-Funding%20Application%20Final.pdf) is a little vague on the scope of this bug, so my assumption is that it's everywhere on the application service.  I'm going to manually re-check all forms **specifically** for the "Label in name" accessibility issue, and make sure it's covered off wherever I find it, manually doing this is a bit tedious and could be error prone though, but due to the nature of screen readers/form runner I'm not sure if there's an easier way to do this.

Also - the document suggestion is a little different from what I've currently done, as in it specifies we should label just the text area, but in doing so screen readers completely skip over the labels/text above the area, so the user loses a lot of context. 

What I've went with (currently in this PR, and seen on the second [[After](https://user-images.githubusercontent.com/117724519/214770064-a8957c9e-8282-44cb-8566-6afdcb7d5574.mp4)] video) is to have all the relevant text content inside of the label, so that the screen reader announces it during tab completion or navigation and gives the user full context - I'm happy to take suggestions on this though, as a lot of the material I've read seems subjective/not granular enough to address this scenario.

### CODE REVIEW

File: form_jsons/public/cy/buddion-cymunedol.json [hunk 1]
```diff
@@ -1,7 +1,9 @@
 {
-  "metadata":{ "isWelsh": true },
+  "metadata": {
+    "isWelsh": true
+  },
   "startPage": "/potensial-i-gyflawni-buddion-cymunedol",
-  "backLinkText": "Yn ôl i'r trosolwg o'r cais",
+  "backLinkText": "Yn \u00f4l i'r trosolwg o'r cais",
```
Comment (tferns) [inline form_jsons/public/cy/buddion-cymunedol.json:6]: I don't think we want this.  Looking into fix.

Comment (tferns) [reply form_jsons/public/cy/buddion-cymunedol.json]: Fixed in https://github.com/communitiesuk/funding-service-design-frontend/pull/249/commits/05079d40f39001bb79552d0348bca94d52a9a17c and https://github.com/communitiesuk/funding-service-design-frontend/pull/249/commits/bbf2e458ef95a7fe2f4d6c37537b8c890cda22df

**Comment by tferns** [comment]
**Note:** I've pushed https://github.com/communitiesuk/funding-service-design-frontend/pull/249/commits/2089d867bde60466f978ea8222f0cc7fef3c4f32 as I originally botched this in https://github.com/communitiesuk/funding-service-design-frontend/pull/249/commits/bbf2e458ef95a7fe2f4d6c37537b8c890cda22df

I'd do an interactive re-base to cleanup the history but since I've already pushed to the remote and sent out the PR I'll just squash the PR (given this actually gets through review stages, might be done differently)

**Comment by gidsg** [comment]
@tferns can you include an example of the generated markup including both the label and the textarea?

We should have something linking the elements e.g. `label for` or `aria-describedby` on the textarea?

**Comment by tferns** [comment]
> @tferns can you include an example of the generated markup including both the label and the textarea?

Good point, here is a snippet of the before HTML:

```html
<div class="govuk-form-group">
   <label class="govuk-label govuk-label--s" for="btTtIb"> Tell us about your organisation's main activities </label>
   <div id="btTtIb-hint" class="govuk-hint">
      <span class="govuk-body">Activity 1</span>
   </div>
   <textarea class="govuk-textarea govuk-js-character-count" id="btTtIb" name="btTtIb" rows="5" aria-describedby="btTtIb-info btTtIb-hint"></textarea>
   <div id="btTtIb-info" class="govuk-hint govuk-character-count__message" aria-live="polite">You have 500 words remaining</div>
</div>
```

And here is the after:

```html
<div class="govuk-form-group">
  <label class="govuk-label" for="btTtIb">
    <span class="govuk-label govuk-label--s">Tell us about your organisation's main activities</span>
    <span class="govuk-body">Activity 1</span>
  </label>
  <textarea class="govuk-textarea govuk-js-character-count" id="btTtIb" name="btTtIb" rows="5" aria-describedby="btTtIb-info"></textarea><div id="btTtIb-info" class="govuk-hint govuk-character-count__message" aria-live="polite">You have 500 words remaining</div>
</div>
```

**Review by gidsg — APPROVED** [review]

**Pull request merged**